ot/mpz-ot/src/kos/sender.rs

```diff
@@ -233,6 +196,64 @@ where
     type Msg = Message<BaseOT::Msg>;
 }
 
+#[async_trait]
+impl<BaseOT> OTSetup for Sender<BaseOT>
+where
+    BaseOT: OTSetup + OTReceiver<bool, Block> + Send,
+{
+    async fn setup<
+        Si: IoSink<Message<BaseOT::Msg>> + Send + Unpin,
+        St: IoStream<Message<BaseOT::Msg>> + Send + Unpin,
+    >(
+        &mut self,
+        sink: &mut Si,
+        stream: &mut St,
+    ) -> Result<(), OTError> {
+        if self.state.is_extension() {
+            return Ok(());
```

Comment:
In what situations will it make sense for KOS to re-use baseOT delta? I cant think of any. Maybe we should only allow calling setup() in state.is_initialized() ?

Reply:
I was trying to point out that there seems to be no situation where calling `setup` twice makes sense.
If the user calls `setup` twice, it is a sign of the user making an error.
That's why I was suggesting to put there:
```rust
if ! self.state.is_initialized() {
            return Err(...);
        }
```

Reply:
Ok, I see the reason.
I do think that calling `setup` twice specifically for KOS15 should be treated as an error.

There is nothing wrong with the dependent protocol first checking if setup is needed before calling it.

We can add a `is_setup_needed()` method to the dependency:
```rust
fn is_setup_needed() -> bool {
    self.state.is_initialized()
}
```

Reply:
adding a `Setup` state doesn't seem to be trivial. if I feel motivated enough, i'll open a new PR for it.